app/helpers/application_helper.rb

```diff
@@ -39,6 +39,14 @@ def facility_product_path(facility, product)
     send(method, facility, product)
   end
 
+  def warning_if_instrument_is_offline_or_partially_available(instrument)
```

Comment:
Good call on pulling this up into a helper. I don't have a strong feeling, but I wonder if `ApplicationHelper` is the right place for this to live. Should we make a PartialAvailabilityHelper?

Reply:
I like `ApplicationHelper` since this method is shared between views of different controllers, so it seems like the textbook place to put! I am not a fan of introducing another module just for this one method here—if, in the future we end up having several helpers related solely to this funcitonality, I could see grouping them into a module then (probably after we have 3 such methods or so), so I’ll leave it as is.